src/Processors/Formats/Impl/Parquet/ParquetRecordReader.cpp

```diff
@@ -0,0 +1,393 @@
+                if (col_descriptor.type_length() <= static_cast<int>(sizeof(Decimal128)))
+                    return makeDecimalLeafReader<Decimal128>();
+                else if (col_descriptor.type_length() <= static_cast<int>(sizeof(Decimal256)))
+                    return makeDecimalLeafReader<Decimal256>();
+            }
+
+            return throwUnsupported(PreformattedMessage::create(
+                ", invalid type length: {}", col_descriptor.type_length()));
+        }
+        default:
+            return throwUnsupported();
+    }
+}
+
+std::unique_ptr<ParquetColumnReader> ColReaderFactory::fromInt32INT(const parquet::IntLogicalType & int_type)
+{
+    switch (int_type.bit_width())
+    {
+        case sizeof(Int32):
+        {
+            if (int_type.is_signed())
+                return makeLeafReader<DataTypeInt32>();
+            else
+                return makeLeafReader<DataTypeUInt32>();
+        }
+        default:
+            return throwUnsupported(PreformattedMessage::create(", bit width: {}", int_type.bit_width()));
+    }
+}
+
+std::unique_ptr<ParquetColumnReader> ColReaderFactory::fromInt64INT(const parquet::IntLogicalType & int_type)
+{
+    switch (int_type.bit_width())
+    {
+        case sizeof(Int64):
+        {
+            if (int_type.is_signed())
+                return makeLeafReader<DataTypeInt64>();
+            else
+                return makeLeafReader<DataTypeUInt64>();
+        }
+        default:
+            return throwUnsupported(PreformattedMessage::create(", bit width: {}", int_type.bit_width()));
+    }
+}
+
+// refer: GetArrowType method in schema_internal.cc of arrow
+std::unique_ptr<ParquetColumnReader> ColReaderFactory::makeReader()
+{
+    // this method should to be called only once for each instance
+    SCOPE_EXIT({ page_reader = nullptr; });
+    assert(page_reader);
+
+    switch (col_descriptor.physical_type())
+    {
+        case parquet::Type::BOOLEAN:
+            break;
+        case parquet::Type::INT32:
+            return fromInt32();
+        case parquet::Type::INT64:
+            return fromInt64();
+        case parquet::Type::INT96:
+        {
+            DataTypePtr read_type = ch_type;
+            if (!isDateTime64(ch_type))
+            {
+                auto scale = getScaleFromArrowTimeUnit(reader_properties.coerce_int96_timestamp_unit());
+                read_type = std::make_shared<DataTypeDateTime64>(scale);
+            }
+            return std::make_unique<ParquetLeafColReader<ColumnDecimal<DateTime64>>>(
+                col_descriptor, read_type, std::move(meta), std::move(page_reader));
+        }
+        case parquet::Type::FLOAT:
+            return makeLeafReader<DataTypeFloat32>();
+        case parquet::Type::DOUBLE:
+            return makeLeafReader<DataTypeFloat64>();
+        case parquet::Type::BYTE_ARRAY:
+            return fromByteArray();
+        case parquet::Type::FIXED_LEN_BYTE_ARRAY:
+            return fromFLBA();
```

Comment:
It appears that the `Bool` and `FLBA (fixed string)` types are not currently supported, Are there any plans to add support for these types in the future?

Reply:
Bool types are distinct due to their 1-bit width,  while clickhouse uses a UInt8 column vector

Reply:
With current framework, supporting these types is not difficult. My plan is to support nested type after this PR, then support all parquet types and enable 02735_parquet_encoder test.

Reply:
@copperybean Any news on boolean support? I was looking at the code and it looks like `ParquetPlainValuesReader` needs to be extended for boolean type. The problem is `ParquetPlainValuesReader` is tightly coupled to `ParquetDataBuffer`, which does not allow single bit reads.

Moreover, `ParquetPlainValuesReader` seems to be templated on the output clickhouse column type. Isn't that a problem? What if there are two distinct parquet types that would result in the same output column type, how would one differentiate? For instance: uint8_t and booleans.

I might be missing something, though. I only had a very superficial read of the code.